### PR TITLE
Add support for the Ferry GraphQL client library

### LINKED ISSUE #12: Add Ferry support
Some users have expressed interest in [Ferry](https://pub.dev/packages/ferry) support, which is a GraphQL client, built on top of the same gql_exec/gql_link framework that `graphql` uses.

### PR DESCRIPTION
**Pull request by shyndman**
Fixes #12 

WIP

### CODE REVIEW

File: packages/nhost_ferry_adapter/example/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: nhost_ferry_adapter_example
+description: Demonstrates usage of the nhost_ferry_adapter package
+publish_to: none
+
+environment:
+  sdk: '>=2.12.0 <3.0.0'
+
+dependencies:
+  ferry:
```
Comment (shyndman) [inline packages/nhost_ferry_adapter/example/pubspec.yaml:9]: comment that constraint will be provided via ferry_adapter

File: packages/nhost_ferry_adapter/lib/postgres_types.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import 'package:built_value/serializer.dart';
+
```
Comment (shyndman) [inline packages/nhost_ferry_adapter/lib/postgres_types.dart:2]: This entire file needs to either be replaced by something more robust, or pulled into its own lib.

File: packages/nhost_ferry_adapter/lib/src/create_client.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  return createNhostFerryClientForAuth(
+    nhostGqlEndpointUrl,
+    nhostClient.auth,
+    gqlCache: gqlCache,
+    defaultHeaders: defaultHeaders,
+    httpClientOverride: httpClientOverride,
+  );
+}
+
+/// Constructs a GQL client for accessing Nhost.io's backend.
```
Comment (shyndman) [inline packages/nhost_ferry_adapter/lib/src/create_client.dart:43]: Mention Ferry

File: packages/nhost_flutter_graphql/example/README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Nhost GraphQL Support for Flutter Examples
 
-* [Simple GraphQL example](https://github.com/nhost/nhost-dart/packages/nhost_flutter_auth/example/lib/simple_graphql_example.dart): Demonstrates establishing a GraphQL connection, and interaction with widgets from the `graphql` package
-* [Todos Quick Start example](https://github.com/nhost/nhost-dart/packages/nhost_flutter_auth/example/lib/todos_quick_start_example.dart): A Flutter implementation of the [Nhost Quick Start app](https://docs.nhost.io/quick-start)
+* [Simple GraphQL example](https://github.com/nhost/nhost-dart/blob/main/packages/nhost_flutter_auth/example/lib/simple_graphql_example.dart): Demonstrates establishing a GraphQL connection, and interaction with widgets from the `graphql` package
```
Comment (shyndman) [inline packages/nhost_flutter_graphql/example/README.md:3]: pull these corrections out into a separate PR

File: packages/nhost_ferry_adapter/example/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Nhost Ferry Adapter Examples
+
+Demonstrates the creation and usage of a Ferry client connected to an Nhost
```
Comment (shyndman) [inline packages/nhost_ferry_adapter/example/README.md:3]: Mention the requirement to run `build_runner` before this is usable

**Comment by FaFre** [comment]
Whats the ETA of this?

**Comment by nunopato** [comment]
closing this PR as it is very outdated
